orchestrator/crates/orchestrator/src/core/client/alert/sns.rs

```diff
@@ -0,0 +1,67 @@
+use async_trait::async_trait;
+use aws_config::SdkConfig;
+use aws_sdk_sns::Client;
+use std::sync::Arc;
+
+use super::AlertError;
+use crate::{core::client::alert::AlertClient, types::params::AlertArgs};
+
+pub struct SNS {
+    pub client: Arc<Client>,
+    pub topic_arn: Option<String>,
+}
+
+impl SNS {
+    pub(crate) fn constructor(client: Arc<Client>) -> Self {
```

Comment:
By convention this should be named `new`.
```suggestion
    pub(crate) fn new(client: Arc<Client>) -> Self {
```

Reply:
Actually have diffrent reason to have this, but it is not longer valid so renamed to new and made the code clean